### ISSUE TITLE
[no-array-constructor] should report error when passing a non-number to Array constructor

### ISSUE THREAD

**Issue by xcatliu** [eslint/eslint#12273]
**What rule do you want to change?**
no-array-constructor

**Does this change cause the rule to produce more or fewer warnings?**
more warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
new default behavior

**Please provide some example code that this change will affect:**

```js
Array('3');
```

**What does the rule currently do for this code?**
No errors

**What will the rule do after it's changed?**
Report an error: don't pass a non-number to Array constructor

**Are you willing to submit a pull request to implement this change?**
Sure, I can do this.

**Comment by xcatliu** [comment]
As [the document](https://eslint.org/docs/rules/no-array-constructor) said:
> Use of the Array constructor to construct a new array is generally discouraged in favor of array literal notation because of the single-argument pitfall and because the Array global may be redefined. The exception is when the Array constructor is used to intentionally create sparse arrays of a specified size by giving the constructor a single numeric argument.

```js
Array('3');  // This will create an array ['3']
// It should considered as an error like examples in document:
Array(0, 1, 2); // Bad, will create an array [0, 1, 2]
```

**Comment by mdjermanovic** [comment]
Hi @xcatliu, thanks for the issue!

Could you please provide more details on the proposal? There is an example with a string literal, should the rule also consider something else as an error when there's exactly one argument?

**Comment by kaicataldo** [comment]
I'm :-1: for this proposal. Since ESLint doesn't have type information, we can't guarantee that we can enforce this. A type system would be much better suited to catching this!

Example:
```js
// file1.js
export const NUM = 3;

// file2.js
import { NUM } from '.file';

Array(NUM); // ESLint has no way of knowing the type of this argument
```
